packages/application-config/src/load-config.ts

```diff
@@ -37,52 +37,97 @@ const loadJsModule: LoaderSync = (filePath) => {
+      `.${configFileName}.js`,
+      `.${configFileName}.cjs`,
+      `.${configFileName}.mjs`,
+      `.${configFileName}.ts`,
+      `${configFileName}.json`,
+      `${configFileName}.js`,
+      `${configFileName}.cjs`,
+      `${configFileName}.mjs`,
+      `${configFileName}.ts`,
+    ],
+    loaders: {
+      noExt: defaultLoaders['.json'],
+      '.js': loadJsModule,
+      '.cjs': loadJsModule,
+      '.mjs': loadJsModule,
+      '.ts': loadJsModule,
+    },
+  });
+};
+
+const customApplicationExplorer = getExplorer('custom-application-config');
+const customViewExplorer = getExplorer('custom-view-config');
+
+// const moduleName = 'custom-application-config';
+// const explorer = cosmiconfigSync(moduleName, {
+//   // Restrict the supported file formats / names
+//   searchPlaces: [
+//     `.${moduleName}rc`,
+//     `.${moduleName}.json`,
+//     `.${moduleName}.js`,
+//     `.${moduleName}.cjs`,
+//     `.${moduleName}.mjs`,
+//     `.${moduleName}.ts`,
+//     `${moduleName}.json`,
+//     `${moduleName}.js`,
+//     `${moduleName}.cjs`,
+//     `${moduleName}.mjs`,
+//     `${moduleName}.ts`,
+
+//     'custom-view.config.json',
+//     'custom-view.config.js',
+//     'custom-view.config.cjs',
+//     'custom-view.config.mjs',
+//     'custom-view.config.ts',
+//   ],
+//   loaders: {
+//     noExt: defaultLoaders['.json'],
+//     '.js': loadJsModule,
+//     '.cjs': loadJsModule,
+//     '.mjs': loadJsModule,
+//     '.ts': loadJsModule,
+//   },
+// });
 
 export const getConfigPath = () => {
-  const configFile = explorer.search();
-  if (!configFile) {
-    throw new Error(
-      `Missing or invalid Custom Application configuration file.`
-    );
+  const customApplicationConfigFile = customApplicationExplorer.search();
+  const customViewConfigFile = customViewExplorer.search();
+
+  if (!customApplicationConfigFile && !customViewConfigFile) {
+    throw new Error(`Missing or invalid configuration file.`);
   }
-  return configFile.filepath;
+  return (
+    customApplicationConfigFile?.filepath || customViewConfigFile?.filepath
+  );
 };
 
 const loadConfig = (
   applicationPath: string
-): JSONSchemaForCustomApplicationConfigurationFiles => {
-  const configFile = explorer.search(applicationPath);
+): {
+  filepath: string;
+  config: JSONSchemaForCustomApplicationConfigurationFiles;
+} => {
+  const customApplicationConfigFile =
+    customApplicationExplorer.search(applicationPath);
+  const customViewConfigFile = customViewExplorer.search(applicationPath);
 
-  if (!configFile || !configFile.config) {
+  if (
+    (!customApplicationConfigFile || !customApplicationConfigFile.config) &&
+    (!customViewConfigFile || !customViewConfigFile.config)
+  ) {
     throw new MissingOrInvalidConfigError(
-      `Missing or invalid Custom Application configuration file.`
+      `Missing or invalid configuration file.`
     );
   }
-  return configFile.config;
+  return (customApplicationConfigFile || customViewConfigFile)!;
```

Comment:
We now return not only the configuration but also the filename so later we can know which one we processed.